### PR TITLE
Adding Slack API endpoint

### PR DESCRIPTION
**Pull request by Brunomachadob**
This is adding an endpoint that replies according to the expected Slack API protocol for slash commands.

With this it will be possible to create Slack apps that are able to respond to `/shouldideploy [tz]` with:
image
image

Or in case of invalid `tz`:
image

The `tz` can be configured via:
1. `req.body.text`: Command text, eg. `/shouldideploy Europe/Berlin`
2. `req.query.tz`: From the command configuration, you can already setup the host with the tz, eg. `https://shouldideploy.today/api/slack?tz=Europe%2FBerlin`
3. `Time.DEFAULT_TIMEZONE`: `UTC`

It is not easy to get the `tz` from the Slack slash command API, so we will need to keep it configurable like that, for whoever is using slack in a single region, they can already attach it on the query param, but users would also still be able to override it.

If whoever uses this with people from multiple zones, users will need to override always or use the default stipulated by the slack workspace. Another option would also be to create multiple slash commands for each common region, e.g 
* `/shouldideploy_eu`
* `/shouldideploy_us`
* ...

I've also added a couple of minor changes to make things more consistent, I'll point them out in the specific places, feel free to ask me to drop/change them.

Example of a Slack command config:
image
*For local development, we must use ngrok to proxy Slack requests to our localhost server*

I guess you can create a simple Slack APP and publish it on their store, I believe no code is needed, you can do it only from the Slack API page, or people could also create inside their own workspace and configure as shown above.

Thanks!

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -50,24 +55,20 @@ class Page extends React.Component {
           <link
             rel="icon"
             type="image/png"
-            href={
-              this.state.now.isFriday()
```
Comment (Brunomachadob) [inline pages/index.js:54]: This was inconsistent with the API, as there, the check is:
```javascript
!time.isFriday() && !time.isWeekend()
```

We shouldn't show a red fav icon during the weekend?

File: pages/index.js [hunk 2]
```diff
@@ -50,24 +55,20 @@ class Page extends React.Component {
...
-                ? 'https://shouldideploy.today/dots-red.png'
-                : 'https://shouldideploy.today/dots.png'
-            }
+            href={shouldIDeployFavIcon(this.state.now)}
             sizes="32x32"
           />
           <meta
             property="og:image"
-            content={
-              this.state.now.isFriday()
```
Comment (Brunomachadob) [inline pages/index.js:63]: Same as above

File: pages/index.js [hunk 3]
```diff
@@ -50,24 +55,20 @@ class Page extends React.Component {
...
-                ? 'https://shouldideploy.today/no.png'
-                : 'https://shouldideploy.today/yes.png'
-            }
+            content={shouldIDeployAnswerImage(this.state.now)}
           />
           <title>Should I Deploy Today?</title>
         </Head>
-        <div className={`wrapper ${this.state.now.isFriday() && 'its-friday'}`}>
```
Comment (Brunomachadob) [inline pages/index.js:70]: Same as above

**Review by baires — APPROVED** [review]

**Comment by baires** [comment]
This looks great! Thanks so much for the hard work here. Im merging this now so we can a nice reminder on slack tomorrow

👏 🥂

**Pull request merged**